Fix debug in Linux

leberus opened the pull request:
This PR aims for fixing debug on Linux. (#7380)

Unfortunately, I did not have so much time lately, so this took more than I expected.

After all these changes, I tested it with some binaries from radare2-regressions/t.archos/Linux-x86_64/

like dbg_fork or dbg_fork2 because that was the main complain.
I could not test all of them due to lack of time, but I plan to do so.

I put the NOT_TO_BE_MERGED_YET just because I want to check a couple of things I wrote down, but should be fast.

Of course, there more people that test debug, the better it will become, otherwise it gets pretty difficult without feedback.

I will update the PR soon and then when all future comments are addressed, we can move on and merge it.

radare commented on libr/core/cmd_debug.c:
```diff
@@ -732,7 +732,13 @@ static void cmd_debug_pid(RCore *core, const char *input) {
 			if (input[2] == '*') {
 				eprintf ("dp %d\n", core->dbg->forked_pid);
 			} else {
+#if __linux__
```
uhm i would prefer not to have os-specifc paths in here

radare commented on libr/core/cmd_debug.c:
```diff
@@ -796,8 +802,16 @@ static void cmd_debug_pid(RCore *core, const char *input) {
 		}
 		break;
 	case '=': // "dp="
+#if __linux__
```
same goes here

radare commented on libr/debug/debug.c:
```diff
@@ -1045,6 +1045,16 @@ R_API int r_debug_continue_kill(RDebug *dbg, int sig) {
+
+		if (reason == R_DEBUG_REASON_NEW_TID) {
+			ret = dbg->tid;
+			if (!dbg->trace_clone)
```
add braces

leberus commented:
I've addressed the comments and removed the NOT_TO_BE_MERGED_YET tag.
I think it can be merged unless something new pops up.

Of course, more people testing it would be appreciated.

radare commented:
lets merge, lgtm, but didnt tested, we have time before the release to check if that breaks mac and windows 

thanks
